src/app/parcels/ParcelsPage.tsx

```diff
@@ -188,6 +189,28 @@ const ParcelsPage: React.FC<{}> = () => {
         }
     }, [parcelId]);
 
+    const clientIdIsValidLink = clientIdForSelectedParcel !== null;
+
+    useEffect(() => {
+        const fetchClientId = async (): Promise<void> => {
+            const { data, error } = await supabase
+                .from("parcels")
+                .select("client_id")
+                .eq("primary_key", parcelId);
+
+            if (error) {
+                console.error("Error fetching parcel data", error.message);
+                return;
+            }
+
+            if (data && data.length > 0) {
```

Comment:
I think typescirpt knows data exists since you've caught the case where error exists
```suggestion
            if (data.length > 0) {
```

Reply:
Changed